app/lib/service/download_counts/download_counts.dart

```diff
@@ -71,17 +71,28 @@ class CountData {
   ///
   /// [patchRangeCounts] has at most [maxRanges] elements and is sorted by
   ///  version ranges.
-  final patchRangeCounts = <VersionRangeCount>[];
+  final List<VersionRangeCount> patchRangeCounts;
 
   /// A list of integers representing the total number of daily downloads of any
   /// version of the package. The list contains at most [maxAge] entries. The
   /// first entry represents the total number of downloads on [newestDate]
   /// followed by the downloads on [newestDate] - 1 and so on.
   ///
   /// Days with no data are represented with `-1`.
-  final totalCounts = List.filled(maxAge, -1, growable: true);
-
-  CountData();
+  final List<int> totalCounts;
+
+  CountData({
+    required this.majorRangeCounts,
+    required this.minorRangeCounts,
+    required this.patchRangeCounts,
+    required this.totalCounts,
+  });
+
+  factory CountData.init() => CountData(
```

Comment:
Why not give it today as the date? life might be simpler :rofl:

Reply:
I am not sure, as discussed. I'll consider it, but not in this change.